requests/models.py

```diff
@@ -404,7 +404,8 @@ def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
         if headers:
-            self.headers = CaseInsensitiveDict(validate_header(i) for i in headers.items())
+            valid_headers = filter(is_valid_header, headers.items())
+            self.headers = CaseInsensitiveDict((to_native_string(name), val) for name, val in valid_headers)
```

Comment:
I mean, `filter` is kind of a lie here (it's really `map`, and even there we aren't transforming the data), so you could have just one loop that calls `is_valid_header` then sets the kvp in the CaseInsensitiveDict.